packages/react-ui/src/components/molecules/UploadField.js

```diff
@@ -0,0 +1,198 @@
+import { Button, IconButton, InputAdornment, TextField } from '@mui/material';
+import makeStyles from '@mui/styles/makeStyles';
+import { Cancel } from '@mui/icons-material';
+
+const useStyles = makeStyles((theme) => ({
+  uploadField: {
+    '& .MuiInputBase-root': {
+      cursor: 'pointer',
+
+      '& input': {
+        cursor: 'pointer'
+      },
+      '&.Mui-disabled': {
+        pointerEvents: 'none',
+
+        '& .MuiButtonBase-root': {
+          color: theme.palette.text.disabled
+        }
+      },
+      '&.MuiInputBase-sizeSmall': {
+        paddingRight: theme.spacing(0.5)
+      }
+    },
+    '& .MuiFormLabel-root': {
+      cursor: 'pointer',
+
+      '&.Mui-disabled': {
+        pointerEvents: 'none'
+      }
+    }
+  },
+
+  focused: {
+    '& .MuiOutlinedInput-root, & .MuiFilledInput-root': {
+      backgroundColor: theme.palette.background.paper
+    },
+    '& .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline, & .MuiInputBase-root.MuiFilledInput-root::after':
+      {
+        opacity: 1,
+        border: `2px solid ${theme.palette.primary.main}`
+      },
+    '& .MuiFormLabel-root': {
+      color: theme.palette.primary.main
+    }
+  }
+}));
+
+function UploadField(props) {
+  const classes = useStyles();
+  const uploadInputRef = useRef(null);
+  const textFieldRef = useRef(null);
+
+  const [filesText, setFilesText] = useState('');
+  const [dragOver, setDragOver] = useState(false);
+
+  useEffect(() => {
+    if (props.files.length === 0) {
+      setFilesText('');
+    } else if (props.files.length === 1) {
+      setFilesText(props.files[0].name);
+    } else {
+      setFilesText(`${props.files.length} files selected`);
+    }
+  }, [props.files]);
+
+  const handleBrowse = () => {
+    uploadInputRef.current?.click();
+  };
+
+  const handleDragOver = (event) => {
+    event.preventDefault();
+    setDragOver(true);
+  };
+
+  const handleDragLeave = (event) => {
+    event.preventDefault();
+    setDragOver(false);
+  };
+
+  const handleDrop = (event) => {
+    event.preventDefault();
+    setDragOver(false);
+
+    const items = event.dataTransfer.items;
+    const newFiles = getAllFiles(items);
+    props.onChange(newFiles);
+  };
+
+  const getAllFiles = (items) => {
+    const newFiles = [];
+    for (let item of items) {
+      if (item.kind === 'file') {
+        const file = item.getAsFile();
+        newFiles.push(file);
+      }
+    }
+
+    return newFiles;
+  };
+
+  const handleFiles = (event) => {
+    const newFiles = Array.from(event.target.files);
+    props.onChange(newFiles);
+  };
+
+  const handleReset = (event) => {
+    event.stopPropagation();
+
+    setFilesText('');
+  };
+
+  const dragPlaceholderText = dragOver ? props.dragPlaceholder : props.placeholder;
+
+  return (
+    <>
+      <TextField
+        {...props}
+        ref={textFieldRef}
+        variant={props.variant}
+        placeholder={dragPlaceholderText}
+        value={filesText}
+        label={props.label}
```

Comment:
All props keeping names should be transmitted with {...props}, right? so I think we can remove:

variant={props.variant}
label={props.label}
helperText={props.helperText}
error={props.error} 

right?

Reply:
I need the placeholder and error because I have to modify them depending on the component is onDrag, the rest you mentioned can be removed, you are right